experimental/packages/proto/src/metrics.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ValueType } from '@opentelemetry/api-metrics';
+import { AggregatorKind, Histogram, MetricKind, MetricRecord, Point } from '@opentelemetry/sdk-metrics-base';
+import { RPCImpl } from 'protobufjs';
+import { hrTimeToLong, toAttributes } from './common';
+import { opentelemetry } from './generated';
+import { Fixed64 } from './types';
+
+export type MetricsClientOptions = {
+    rpcImpl: RPCImpl,
+    startTime: Fixed64,
+};
+export class MetricsServiceClient {
+    private _service: opentelemetry.proto.collector.metrics.v1.MetricsService;
+    private _startTime: Fixed64;
+
+    constructor(options: MetricsClientOptions) {
+        this._service = new opentelemetry.proto.collector.metrics.v1.MetricsService(options.rpcImpl);
+        this._startTime = options.startTime ?? Date.now() * 1000;
+    }
+
+    async export(metricRecords: MetricRecord[]): Promise<unknown> {
+        const request = createExportMetricsServiceRequest(metricRecords, this._startTime);
+        if (!request) return null;
+        return this._service.export(request);
+    }
+}
+
+export function createExportMetricsServiceRequest(metricRecords: MetricRecord[], startTime: Fixed64): opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest | null {
+    if (metricRecords.length === 0) {
+        return null;
+    }
+
+    const resource = metricRecords[0].resource;
+
+    return opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest.fromObject({
```

Comment:
IIUC, we don't need to convert the data to concrete protobuf objects as they are transferred by JSON. These conversion helpers only create data objects with valid shapes according to the proto definitions, like https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/exporter-trace-otlp-http/src/transform.ts#L171.

We can still use the types generated by `pbts` as `opentelemetry.proto.collector.metrics.v1.IExportMetricsServiceRequest` (by adding leading `I`).

Reply:
You're right we don't, but it doesn't make sense to duplicate all the transformations. We can use the interfaces and generate valid JSON objects then pass those to `fromObject` in order to get protobuf or we can use `fromObject` and call `toJSON` at the end to get JSON back out. Either way is fine with me.

Reply:
Actually it looks like even the interfaces can't be used directly for the JSON representation of the protobuf. The interfaces generated use enums, which when serialized directly with JSON.stringify are converted to numbers and not the string representation required by [JSON mapping](https://developers.google.com/protocol-buffers/docs/proto3#json). So in order to map correctly, we would need to either manually fix that (and create our own types i guess?) or use `fromObject` then `toJSON` anyway.